File: src/boiler/erlang/day4.erl
-module(day4).
-export([day_four/0]).

print_young() ->
    io:fwrite("You are young~n", []).

print_old() ->
    io:fwrite("You are old~n", []).

print_teenager() ->
    io:fwrite("You are a teenager~n", []).

print_error() ->
    io:fwrite("ERROR: This person is not valid, setting age to 0.~n", []).

age(Age) when Age < 0 ->
    print_error();
age(Age) when Age >= 0, Age < 13 ->
    print_young();
age(Age) when Age >= 13, Age < 18 ->
    print_teenager();
age(_) ->
    print_old().

day_four() ->
    {ok, [Age]} = io:fread("enter age", "~d"), 
    io:fwrite("Age is ~w~n", [Age]),
    age(Age).

File: src/boiler/elixir/day4.exs
defmodule Day4 do
    def my_age(age) when
        age < 0 do
            IO.puts("Negative age is invalid, please resubmit!")
            day4()
    end
    def my_age(age) when
        age < 13 do
            IO.puts("You are young")
    end 
    def my_age(age) when
        age < 18 and age > 12 do
            IO.puts("You are a teenager")
    end
    def my_age(_) do
        IO.puts("You are old")
    end
    def day4 do
        age = String.to_integer(String.trim(IO.gets("Enter your age, please! ")))
        my_age(age)
    end
end
